Dockerfile

```diff
@@ -25,27 +43,51 @@ ARG PKG_SRC
 ARG PKG_NAME
 
 WORKDIR /app
+
+ENV PIP_NO_COMPILE=1 \
+    PIP_DISABLE_PIP_VERSION_CHECK=1
+
+# Install build tools to compile dependencies that don't have prebuilt wheels
+RUN apk add --update --no-cache build-base git poetry libffi-dev
+RUN pip install --no-cache-dir --upgrade pip setuptools wheel
+
+# Copy the bare minimum needed for specifying dependencies.
+# This will enable better layer caching and faster builds when iterating locally.
+# `--without-hashes` is used to ensure the `pip` cache mount is used for packages that
+# would otherwise only match the sdist and therefore have to be built for every run.
+# References:
+#   * https://pythonspeed.com/articles/pipenv-docker/
+#   * https://hub.docker.com/r/docker/dockerfile
+COPY pyproject.toml poetry.lock ./
+RUN poetry export --without-hashes --format requirements.txt --output requirements.txt
+
+# Cache the pip installed dependencies
+# NOTE: This `--mount` feature requires BUILDKIT to be used
+RUN --mount=type=cache,id=pip,target=/root/.cache/pip \
+    set -eux; \
+    pip cache info; \
+    pip cache list; \
+    pip install --user -r requirements.txt
```

Comment:
Understood...and don't disagree. The reason for this format is due to the initial effort to emulate what is done in the [Dockerfile for the `python:3.10-alpine`](https://github.com/docker-library/python/blob/f7bb6f4ee29f0cdc8c1dbb55b17aa4e0f38497e0/3.10/alpine3.16/Dockerfile#L26) image this work is based on. I'm not terribly familiar with Docker (even still) and assumed what was good for Python was good here.

Reply:
Yep. I'm fine with using this style. Looking at the history of that Python image because I was curious, it looks like one of the docker-library maintainers adds `set -eux` anytime he splits a `RUN` command into multiple lines (e.g., [this commit](https://github.com/docker-library/python/commit/c99c77547e99f80fb9b895b7dc13b88b78170e2e))